ops/main.py

```diff
@@ -37,9 +48,37 @@ def main(charm_class: Type[ops.charm.CharmBase], use_juju_for_storage: Optional[
 
     See `ops.main() <#ops-main-entry-point>`_ for details.
     """
-    warnings.warn(
-        'Calling `ops.main.main()` is deprecated, call `ops.main()` instead',
-        DeprecationWarning,
-        stacklevel=2,
+    # Normally, we would do warnings.warn() with a DeprecationWarning, but at
+    # this point in the charm execution, the framework has not been set up, so
+    # we haven't had a chance to direct warnings where we want them to go. That
+    # means that they'll end up going to stderr, and with actions that means
+    # they'll end up being displayed to the user.
+    # This means that we need to delay emitting the warning until the framework
+    # has been set up, so we wrap the charm and do it on instantiation. However,
+    # this means that a regular warning call won't provide the correct filename
+    # and line number - we could wrap this in a custom formatter, but it's
+    # simpler to just use a logging.warning() call instead.
+    # Note that unit tests almost never call main(), so wouldn't show this
+    # warning, so we're relying on integration tests, which aren't going to be
+    # treating warning calls in any special way. That means that we can just
+    # output the warning directly as a log message, and the effect will be the
+    # same.
+    # Note also that this will be logged with every event. Our assumption is
+    # that this will be noticeable enough during integration testing that it
+    # will get fixed before going into production.
```

Comment:
I'd like to see the assumptions in this comment validated, and the comment rewritten to maybe 1/3 of the size.

Reply:
Hey - maybe you missed the comment on the PR: I've switched it to draft while I adjust some things, and no need to review until then (moving to draft doesn't unsubscribe so de-selecting reviewers doesn't really do anything unfortunately).

Reply:
Sorry for the early fly-by, I'm cleaning up my github backlog 🙇🏻 

In any case, I wanted to understand the 2 alternatives in the PR description.